runtime/stdlib/account.go

```diff
@@ -2272,22 +2344,220 @@ func newAuthAccountStorageCapabilitiesIssueFunction(
 				panic(errors.NewUnreachableError())
 			}
 
-			// TODO: create and write StorageCapabilityController
+			// Create and write StorageCapabilityController
 
 			borrowStaticType := interpreter.ConvertSemaReferenceTypeToStaticReferenceType(gauge, borrowType)
 
+			var capabilityID uint64
+			var err error
+			errors.WrapPanic(func() {
+				capabilityID, err = idGenerator.GenerateAccountID(address)
+			})
+			if err != nil {
+				panic(err)
+			}
+
+			capabilityIDValue := interpreter.UInt64Value(capabilityID)
+
+			controller := interpreter.NewStorageCapabilityControllerValue(
+				gauge,
+				borrowStaticType,
+				targetPathValue,
+				capabilityIDValue,
+			)
+
+			storeCapabilityController(inter, address, capabilityIDValue, controller)
+			recordPathCapabilityController(inter, locationRange, address, targetPathValue, capabilityIDValue)
+
 			return interpreter.NewStorageCapabilityValue(
 				gauge,
-				// TODO:
-				interpreter.TodoCapabilityID,
+				capabilityIDValue,
 				addressValue,
-				pathValue,
+				targetPathValue,
 				borrowStaticType,
 			)
 		},
 	)
 }
 
+// CapabilityControllerStorageDomain is the storage domain which stores
+// capability controllers by capability ID
+const CapabilityControllerStorageDomain = "cap_con"
```

Comment:
If these are new domains, is it possible to create a `Path` value with this domain? We should have some tests to make sure these can't "escape" into user code.

Reply:
Good point! We already have other storage domains which should not be user accessible, like for the capability inbox. I'll add tests if they don't already exist.

Reply:
Added tests in the PR feedback PR: [`fd60f78` (#2495)](https://github.com/onflow/cadence/pull/2495/commits/fd60f78532e32bfd38d7b343ac50f40c0f71971d)